File: frontend/src/types/prediction.ts
export interface PredictionRequest {
  location: string;
  carpet_area?: number | null;
  floor?: number | null;
  transaction?: string | null;
  furnishing?: string | null;
  facing?: string | null;
  overlooking?: string | null;
  bathroom?: number | null;
  balcony?: number | null;
  ownership?: string | null;
  super_area?: number | null;
  total_floors?: number | null;
  parking_spaces?: number | null;
  is_covered?: number | null;
}

export interface PredictionResponse {
  predicted_price: number;
}

File: backend/app/schemas/prediction.py
from typing import Optional
from pydantic import BaseModel


class PredictionRequest(BaseModel):
    location: str
    carpet_area: Optional[float] = None
    floor: Optional[float] = None
    transaction: Optional[str] = None
    furnishing: Optional[str] = None
    facing: Optional[str] = None
    overlooking: Optional[str] = None
    bathroom: Optional[float] = None
    balcony: Optional[float] = None
    ownership: Optional[str] = None
    super_area: Optional[float] = None
    total_floors: Optional[float] = None
    parking_spaces: Optional[float] = None
    is_covered: Optional[float] = None


class PredictionResponse(BaseModel):
    predicted_price: float